src/main/java/org/wso2/carbon/module/swiftiso20022/utils/MT940ValidationUtils.java

```diff
@@ -0,0 +1,404 @@
+                closeAvailBalance = line;
+            } else if (line.startsWith(ConnectorConstants.MT940_FORWARD_AVAIL_BAL)) {
+                forwardAvailBalance = line;
+            }
+        }
+
+        if (!openBalance.substring(13, 15).equals(closeBalance.substring(13, 15))) {
+            return false;
+        } else if (closeAvailBalance != null &&
+                !openBalance.substring(13, 15).equals(closeAvailBalance.substring(12, 14))) {
+            return false;
+        }
+
+        return forwardAvailBalance == null ||
+                openBalance.substring(13, 15).equals(forwardAvailBalance.substring(12, 14));
+    }
+
+    /**
+     * Method to validate the MT940 format
+     *
+     * @param fields  Fields in MT940 format
+     * @return    ErrorModel MT940 format is invalid
+     */
+    public static ErrorModel validateMT940Format(Map<String, Object> fields) {
+        ErrorModel errorModel = new ErrorModel();
+
+        errorModel = validateTransactionReference((String) fields.get(ConnectorConstants.TRANSACTION_REF));
+        if (errorModel.isError()) {
+            return errorModel;
+        }
+
+        errorModel = validateRelatedReference((String) fields.get(ConnectorConstants.RELATED_REF));
+        if (errorModel.isError()) {
+            return errorModel;
+        }
+
+        errorModel = validateAccountIdentifier((String) fields.get(ConnectorConstants.ACC_IDENTIFICATION));
+        if (errorModel.isError()) {
+            return errorModel;
+        }
+
+        errorModel = validateStatementNumber((String) fields.get(ConnectorConstants.STATEMENT_NUMBER));
+        if (errorModel.isError()) {
+            return errorModel;
+        }
+
+        errorModel = validateOpeningBalance((String) fields.get(ConnectorConstants.OPENING_BALANCE));
+        if (errorModel.isError()) {
+            return errorModel;
+        }
+
+        errorModel = validateClosingBalance((String) fields.get(ConnectorConstants.CLOSING_BALANCE));
+        if (errorModel.isError()) {
+            return errorModel;
+        }
+
+        errorModel = validateClosingAvailableBalance((String) fields
+                .get(ConnectorConstants.CLOSING_AVAIL_BALANCE));
+        if (errorModel.isError()) {
+            return errorModel;
+        }
+
+        errorModel = validateForwardAvailableBalance((String) fields
+                .get(ConnectorConstants.FORWARD_CLOSING_AVAIL_BALANCE));
+        if (errorModel.isError()) {
+            return errorModel;
+        }
+        return errorModel;
+    }
+
+    /**
+     * Method to validate whether Transaction reference is valid.
+     *
+     * @param transactionRef  Transaction reference to be validated
+     * @return    Whether transaction reference is valid
+     */
+    public static ErrorModel validateTransactionReference(String transactionRef) {
+        ErrorModel errorModel = new ErrorModel();
+        String reference = transactionRef.split(ConnectorConstants.COLON)[2];
```

Comment:
Possible ArrayIndexOutOfBoundsException. Applicable for other places as well

Reply:
Fixed in https://github.com/wso2-extensions/mediation-swift-iso20022-module/pull/2/commits/8d6b87e8988955fbe9679e7afa9d4285b0ff0ac8